pytorch_lightning/plugins/native_amp.py

```diff
@@ -69,6 +69,11 @@ def optimizer_step(self, trainer, optimizer, closure):
         # TODO: pass the closure to the step ASAP
         with trainer.profiler.profile("closure"):
             closure()
+
+        if not self.trainer.train_loop.automatic_optimization:
+            trainer.scaler.unscale_(optimizer)
+            trainer.call_hook("on_after_backward")
```

Comment:
does it mean that this hook is called only in manual optim?

Reply:
Hey both. 
For automatic_optimization, it is fine. It is called on training_step_and_backard.

For manual_optimization, we can't predict when people are going to make a `.step`. Therefore, the hook needs to be call after latest closure and `.step`.

However, I can move this to the precision plugin to reduce code.

Reply:
This will be taken care in a next PR @awaelchli. I will need to clean out training_loop for `automatic_optimization` first, so it uses the same API as manual one.